clio-util/src/test/scala/org/broadinstitute/clio/util/json/StateDecoderSpec.scala

```diff
@@ -40,8 +40,16 @@ class StateDecoderSpec
     }
   }
 
+  it should "decode a field with an option explicitly set to null" in {
+    implicit val decoder: Decoder[TestClass] =
+      Decoder.fromState(StateDecoder[TestClass].state)
+
+    decode[TestClass]("""{"fieldA": null, "fieldB": 123}""").right.value should be(
```

Comment:
Without the fix in the `StateDecoder`, this would raise an error saying `fieldA` was an extra key.